src/entities/GreenSpinyEgg.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import type { Entity } from './types';
+import { TILE_SIZE } from '../tiles/constants';
+import { ANY_OBJECT_SET } from './constants';
+import { parseSimpleSprite } from './util';
+
+const graphicSets = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
+
+const GreenSpinyEgg: Entity = {
```

Comment:
can you name it SpinyEggGreen and name the file SpinyEggGreen.tsx? Then when we add the orange one, we'll call it SpinyEggOrange, and the good thing is they end up next to each other in the filesystem. It basically groups similar entities together in a primitive way.

I know a lot of entities don't do this, those are older entities. I switched to the "adjective goes to the end" later. We can't just change entity names, as that would break people's levels. So until full binary parsing is done, I'm just leaving the old names.

There is a way to write a converter that would fix people's levels if we did do a name change, but it's such a pain I avoid it as much as possible.

Reply:
well i don't know if we're gonna add the orange spiny egg, for some reason when added it uses the red palette instead of the yellow/orange palette, but yeah i can do that for organization purposes and also in case we do that

Reply:
cool. yeah in general names should have adjectives at the end, even if there's only one. eventually all names will get set to that. like in 2028.

Reply:
I'll add that to the dev guide too